tokio/src/io/mod.rs

```diff
@@ -93,7 +93,8 @@
 //! ```
 //!
 //! [`BufWriter`] doesn't add any new ways of writing; it just buffers every call
-//! to [`write`](crate::io::AsyncWriteExt::write):
+//! to [`write`](crate::io::AsyncWriteExt::write). However, with [`BufWriter`] you
+//!  should be mindful of flushing its contents before it is dropped:
```

Comment:
Perhaps we could use a simpler sentence?

> You must flush the `BufWriter` to ensure that any buffered data is written.

Reply:
Sure, sounds good.